src/Mod/Draft/draftmake/make_copy.py

```diff
@@ -0,0 +1,211 @@
+# ***************************************************************************
+# *   Copyright (c) 2009, 2010 Yorik van Havre <yorik@uncreated.net>        *
+# *   Copyright (c) 2009, 2010 Ken Cline <cline@frii.com>                   *
+# *   Copyright (c) 2020 FreeCAD Developers                                 *
+# *                                                                         *
+# *   This program is free software; you can redistribute it and/or modify  *
+# *   it under the terms of the GNU Lesser General Public License (LGPL)    *
+# *   as published by the Free Software Foundation; either version 2 of     *
+# *   the License, or (at your option) any later version.                   *
+# *   for detail see the LICENCE text file.                                 *
+# *                                                                         *
+# *   This program is distributed in the hope that it will be useful,       *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with this program; if not, write to the Free Software   *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+"""This module provides the code for Draft make_copy function.
+"""
+## @package make_copy
+# \ingroup DRAFT
+# \brief This module provides the code for Draft make_copy function.
+
+import FreeCAD as App
+
+import draftutils.utils as utils
+import draftutils.gui_utils as gui_utils
+
+from draftobjects.rectangle import Rectangle
+from draftobjects.point import Point
+from draftobjects.dimension import LinearDimension
+from draftobjects.wire import Wire
+from draftobjects.circle import Circle
+from draftobjects.polygon import Polygon
+from draftobjects.bspline import BSpline
+from draftobjects.block import Block
+
+
+if App.GuiUp:
+    from draftviewproviders.view_base import ViewProviderDraft
+    from draftviewproviders.view_base import ViewProviderDraftPart
+    from draftviewproviders.view_rectangle import ViewProviderRectangle
+    from draftviewproviders.view_point import ViewProviderPoint
+    from draftviewproviders.view_dimension import ViewProviderLinearDimension
+    from draftviewproviders.view_wire import ViewProviderWire
+
+
+def make_copy(obj, force=None, reparent=False):
```

Comment:
@vocx-fc I read now we have a Document.copyObject() API from @realthunder.
Perhaps we could adopt it also in Draft and deprecate this function?

Reply:
Sure, but maybe leave it for later when the re-organization is done. Make a note about it `TODO: use copyObject instead`.

Making critical changes at this point may cause problems for some tools. So, unless you have already tested the function, I wouldn't do it right now. First ask yourself, how many objects use `make_copy`? And if we change them to use `copyObject`, do the objects work the same?

Reply:
Of course, mine was a reminder for later. (I think there are tons of other old things in Draft that after splitting is complete we could address I think)